src/PAModel/CanvasDocument.cs

```diff
@@ -197,13 +197,59 @@ private static void Wrapper(Action worker, ErrorContainer errors)
                 }
             }
         }
-
+                
         internal CanvasDocument()
         {
             _editorStateStore = new EditorStateStore();
             _templateStore = new TemplateStore();
         }
 
+        internal CanvasDocument(CanvasDocument other)
```

Comment:
In theory, is this the same as 

   CanvasDocument.LoadFromMsApp(other.SaveToMsapp())

Reply:
Yeah, should be.